### ISSUE TITLE
Solved: Collect mails with IMAP IDLE and deliver them to mailcow with LMTP (Getmail for Mailcow)

### ISSUE THREAD

**Issue by christianbur** [mailcow/mailcow-dockerized#1554]
I was looking for a solution to collect mails per IMAP IDLE  from other mailboxes and deliver them to mailcow. However, there is currently no solution to send email via LMTP (port 24) to Deovecot, so I built a solution that works with any number of email accounts. Since the mails are delivered via LMTP, the emails can also be filtered with dovecot sieve. 
In each transmitted email the field "X-getmail-retrieved-from-mailbox" is added in the header, as value it contains the mailbox name from the settings.ini. This field can also be used with sieve. 
Currently i don't delete the mails in the source mailbox yet i just move them to the getmail folder.
This solution runs now for a few months without problems.

https://github.com/christianbur/docker_projects/tree/master/getmail

**Comment by mkuron** [comment]
We already have imapsync integrated into Mailcow, but it doesn't use LMTP (so no Sieve) and polls instead of using IDLE. So there is value in solution, but it would need a tighter integration into Mailcow before we could consider incorporating it.

**Comment by christianbur** [comment]
The existing "imapsync" from mailcow can only be used for an initial migraiton (an entire mailbox at once). My solution is intended for continuous synchronization, because with IMAP IDEL the emails are immediately retrieved from the source mailbox (no delay) and after delivery they can also be filtered or automatically moved by sieve into an  folder. Therefore imapsync and getmail have different tasks.

**Comment by Braintelligence** [comment]
Correct me if I'm wrong, but I was under the impression, that imapsync is triggered not only once but in set time intervals?

**Comment by christianbur** [comment]
is correct, but you always have to wait for the interval. This is not acceptable to me.

**Comment by mkuron** [comment]
So they do fulfill the same purpose. We are not going to add both, but if you can enhance your script to  cover everything imapsync currently does, we could consider replacing imapsync with it.

**Comment by tkorves** [comment]
As it's marked solved, I'll close it.

**Comment by mkuron** [comment]
@tkorves, I think it only says "Solved" because @christianbur has a solution. Having IDLE support in imapsync/getmail is a legitimate feature request, so let's keep it open for now.

**Comment by xpcone** [comment]
In addition his Solution works fine for one receiving address, not for many

**Comment by christianbur** [comment]
Why?
You can override every DEFAULT entry in a specific section.
If you want you can also remove the DEFALT section and write all entries into the specific sections. 

[mail_example.de]
imap_hostname:     mail.example.de
imap_port:         993
imap_username:     mail@example.de
imap_password:     PASSWORD
imap_debug:        False
imap_move_folder:  getmail
lmtp_hostname:     dovecot
lmtp_port:         24
lmtp_recipient:    mail2@example.de
lmtp_debug:        False

**Comment by xpcone** [comment]
There are two questions left:
All in the same ini file?
How do i split / define to which of the 5 destination Mailbox the mail will go.

**Comment by christianbur** [comment]
All in the same ini file! (https://wiki.python.org/moin/ConfigParserExamples)

imap_username is the source imap account (for example gmail) and lmtp_recipient is you mailcow imap account.

**Comment by smjaberl** [comment]
Well, this would still be a very nice feature... 
Maybe this could be merged into the mailcow documentation.

**Comment by christianbur** [comment]
You don't have to change Mailcow for my Getmail, it's just an addon :-)

https://github.com/christianbur/docker_projects/tree/master/getmail

functions:
 - Python 3
 - IMAP IDLE
- Move or delete the messages in the source IMAP folder.
- You can query as many accounts as you want.  
- No change in mailcow necessary
- Since new emails are delivered via LMTP, they can also be filtered with Sieve.

**Comment by smjaberl** [comment]
thanks! I will keep it in mind, but jet I have a setup running with imapsync.

**Comment by andryyy** [comment]
Thank you, @christianbur 

We should at least at that to the docs.

**Comment by byniow** [comment]
> You don't have to change Mailcow for my Getmail, it's just an addon :-)
> 
> https://github.com/christianbur/docker_projects/tree/master/getmail
> 
> functions:
> 
>     * Python 3
> 
>     * IMAP IDLE
> 
>     * Move or delete the messages in the source IMAP folder.
> 
>     * You can query as many accounts as you want.
> 
>     * No change in mailcow necessary
> 
>     * Since new emails are delivered via LMTP, they can also be filtered with Sieve.

How this use? I'm new in docker, mailcow works, but instructions for this plugin is completly not clear.

I'll try put settings.ini in different places without success. Still get

2022-04-18 12:03:54,775 - MainThread - ERROR: ERROR settings.ini not found!
Traceback (most recent call last):
  File "./getmail_imap2lmtp.py", line 309, in <module>
    start_getmail()
  File "./getmail_imap2lmtp.py", line 259, in start_getmail
    for config_name in configparser_file.sections():
AttributeError: 'NoneType' object has no attribute 'sections'

I uncompress archive to /opt/mailcow-dockerized/getmail, compose created directory /opt/mailcow-dockerized/getmail/conf/settings.ini, and i put file inside with result: /opt/mailcow-dockerized/getmail/conf/settings.ini/settings.ini. But docker service still not see it.

**Comment by s4069b** [comment]
byniow - if you have not already worked it out - the Installation instructions are here - https://github.com/christianbur/docker_projects/blob/master/getmail/INSTALL.TXT

Put getmail in /opt/getmail (not in /opt/mail-cow-dockerized/getmail) and then follow the rest of the instructions, your settings.ini should stay in /opt/getmail/conf/settings.ini

I also needed to run another command after setting up the bridge network: "docker network connect network-getmail  mailcowdockerized_dovecot-mailcow_1"
I verified the connection with: "docker network inspect network-getmail"
And then made sure the default section in 'settings.ini' included this line "lmtp_hostname:     mailcowdockerized_dovecot-mailcow_1"

**Comment by byniow** [comment]
EDIT
Thank You very much for answer. I have a lot problems to correct start getmail service. There are more probmems with docker than with getmail. So, AFTER place files in correct localizations we NEED remove ealier gatmail image by docker rm getmail_getmail_1 command and then docker-compose up -d. Now i have info about opened settings.ini. 

So, now works. Thanks again.

**Comment by byniow** [comment]
Sorry for maybye stupid question but i try find solution and can not. I'm not know almost anything about docker, online tutorials so for inly easy configs and i not understand how network connection work. 

Getmail start, i config gmail account and get:

```
2022-06-05 17:15:24,391 - MainThread - INFO: use config file: ./settings.ini
getmail_1  | 2022-06-05 17:15:29,393 - Thread-xxx@gmail.com - INFO: Start Getmail - server: imap.gmail.com:993, username: xxx@gmail.com, ssl: True
getmail_1  | 2022-06-05 17:15:30,578 - Thread-xxx@gmail.com - INFO: Login - status: b'xxx@gmail.com authenticated (Success)'
getmail_1  | 2022-06-05 17:15:30,750 - Thread-xxx@gmail.com - INFO: IMAP fetch mail - inital
root@mail:/opt/getmail# 

```
And nothing more. I not see any nails in mailcow. settings.ini
```
[DEFAULT]
imap_hostname:     10.1.0.20
imap_port:         993
imap_username:     mail@mail.local
imap_password:     xxx
imap_debug:        True
imap_move_enable:  False
imap_move_folder:  getmail
imap_sync_folder:  INBOX
imap_ssl:          True
lmtp_hostname:     mailcowdockerized_dovecot-mailcow_1
lmtp_port:         24
lmtp_recipient:    mail@mail.local
lmtp_debug:        True

#[example_insecure_test1_gmx.de]
#imap_hostname:     imap.gmx.net
#imap_port:         143
#imap_username:     example-test1@gmx.de
#imap_password:     xxxxx
#imap_ssl:          False

#[example_test2_gmx.de]
#imap_hostname:     imap.gmx.net
#imap_username:     example-test2@gmx.de
#imap_password:     xxxxx

[byniow@gmail.com]
imap_hostname:     imap.gmail.com
imap_username:     xxx@gmail.com
imap_password:     xxx 
```

Propably imap hostname is wrong because getmail is connected via bridge to mailcow network. But how find correct address? IP what i set is ip of lan interface where i can connect to mailcow gui, login and password i entered the same like for email account in sogo.

**Comment by byniow** [comment]
Ok, getmail works (read serrings.ini) but not retreive mails. I'll try all day today googling solution without success. 
```
root@mail:/opt/getmail# docker-compose logs
Attaching to getmail_getmail_1
getmail_1  | 2022-06-05 19:39:41,878 - MainThread - INFO: use config file: ./settings.ini
getmail_1  | 2022-06-05 19:39:46,879 - Thread-xxx@gmail.com - INFO: Start Getmail - server: imap.gmail.com:993, username: xxx@gmail.com, ssl: True
getmail_1  | 2022-06-05 19:39:47,973 - Thread-xxx@gmail.com - INFO: Login - status: b'xxx@gmail.com authenticated (Success)'
getmail_1  | 2022-06-05 19:39:48,172 - Thread-xxx@gmail.com - INFO: IMAP fetch mail - inital
```
settings.ini
```
DEFAULT]
imap_port:         993
imap_debug:        True
imap_move_enable:  False
imap_move_folder:  getmail
imap_sync_folder:  INBOX
imap_ssl:          True
lmtp_hostname:     mailcowdockerized_dovecot-mailcow_1
lmtp_port:         24
lmtp_recipient:    mail@mail.local
lmtp_debug:        True

#[example_insecure_test1_gmx.de]
#imap_hostname:     imap.gmx.net
#imap_port:         143
#imap_username:     example-test1@gmx.de
#imap_password:     xxxxx
#imap_ssl:          False

#[example_test2_gmx.de]
#imap_hostname:     imap.gmx.net
#imap_username:     example-test2@gmx.de
#imap_password:     xxxxx

[byniow@gmail.com]
imap_hostname:imap.gmail.com
imap_username:xxx@gmail.com
imap_password:xxx
```
No new mails at all. I reinstalled mailcow from scratch, try many stupid entries - no success....

**Comment by christianbur** [comment]
Since getmail is not part of mailcow you should rather open a ticket here. https://github.com/christianbur/docker_projects/issues

I also monitor two Google IMAP accounts on the imap.gmail.com server, so the script shouldn't  be the problem. 
The "lmtp_hostname" is wrong in my opinion, see here   https://github.com/christianbur/docker_projects/issues/9

**Comment by byniow** [comment]
Yes, s4069b wrote about this, and

```
root@mail:~# docker network inspect network-getmail
[
    {
        "Name": "network-getmail",
        "Id": "c118c1279463b908115f972253095efea7c83c22c8162ede4761f7aa6b91871d",
        "Created": "2022-06-05T16:52:02.786008695Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": true,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.23.11.0/24",
                    "Gateway": "172.23.11.1"
                },
                {
                    "Subnet": "fdcb:9c90:23:11::/64",
                    "Gateway": "fdcb:9c90:23:11::1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "13da277d7cc4d325013c818c632700cc51d09184548c57cb09ef5d82b49c5e69": {
                "Name": "getmail_getmail_1",
                "EndpointID": "8aea13eb75e4b89c883c076ec6c69eb49eacf7272d04beafbcb062892735747d",
                "MacAddress": "02:42:ac:17:0b:02",
                "IPv4Address": "172.23.11.2/24",
                "IPv6Address": "fdcb:9c90:23:11::2/64"
            },
            "4985fd377760e2823b141616c26e9ed15cd2bdd0a46687a0ad448be984ea9022": {
                "Name": "mailcowdockerized_dovecot-mailcow_1",
                "EndpointID": "3f24853965460dd325ea831bf03b22814530af9d862aeb9d5ac4c548b418e504",
                "MacAddress": "02:42:ac:17:0b:03",
                "IPv4Address": "172.23.11.3/24",
                "IPv6Address": "fdcb:9c90:23:11::3/64"
            }
        },
        "Options": {
            "com.docker.network.bridge.name": "br-getmail"
        },
        "Labels": {}
    }
]
```
So, name should fit. To make this i used command from documentation

`docker network create --driver=bridge --ipv6 --subnet=fdcb:9c90:23:11::/64 --gateway=fdcb:9c90:23:11::1 --subnet=172.23.11.0/24 --gateway=172.23.11.1 -o "com.docker.network.bridge.name"="br-getmail" network-getmail`

and then

`docker network connect network-getmail mailcowdockerized_dovecot-mailcow_1`

**Comment by christianbur** [comment]
The command "docker network connect network-getmail mailcowdockerized_dovecot-mailcow_1" doesn't make sense to me, because the connection (container: dovecot-mailcow <-> network: network-getmail) comes through the ../mailcow-dockerized/docker-compose.override.yml. 

see https://github.com/christianbur/docker_projects/issues/9

**Comment by byniow** [comment]
Ok, propably i not understand this link. But - germail get diffrent network, outside mailcow network. How getmail connect between diffrent networks without routing/bridges?

**Comment by christianbur** [comment]
Due to the entries in the [docker-compose.override.yml.](https://github.com/christianbur/docker_projects/blob/master/getmail/mailcow-dockerized_docker-compose.override.yml)

**Comment by byniow** [comment]
Thank You for help. But this solution not work, propably i misfired trivial step, but i reinstall mailcow twice, make step-by-step Your instruction (without s4069b steps) and still getmail is stop on `INFO: IMAP fetch mail - inital`. Thanks for Your effort, i appreciate it. But I give up, I stay with iredmail + fetchmail solution without docker... Much easier i can diagnose problems...

**Comment by s4069b** [comment]
Sorry for the wrong advice with "docker network connect network-getmail mailcowdockerized_dovecot-mailcow_1" 
christianbur corrected me by pointing me back to docker-compose.override.yml. I had copied it into the mailcow directory, but accidentally named that file docker-compose-override.yml (yep, tiny typo - instead of .) So mialcow was not connecting to the bridge network. Renamed the file and it all works fine. I had to go into settings.ini and restore lmtp_hostname: dovecot-mailcow. Stop getmail (change into getmail directory and docker-compose stop) then restart it (docker-compose up -d)
If it's any help, this is how I have managed to re-do my mailcow installation from scratch when I've made a mess of it. 
in this directory  /opt/mailcow-dockerized and as sudo
1. docker-compose stop
2. docker-compose down -v --rmi all --remove-orphans
then, when it's all finished, start over with docker-compose up -d 

getmail is working nicely with gmail now

**Comment by byniow** [comment]
Ok, maybye we try compare config files to find where is problem?
```
root@mail:/opt/getmail# cat conf/settings.ini 
[DEFAULT]
#imap_hostname:     MUST_BE_DEFINED_INDIVIDUALLY
#imap_username:     MUST_BE_DEFINED_INDIVIDUALLY
#imap_password:     MUST_BE_DEFINED_INDIVIDUALLY
imap_port:         993
imap_debug:        True
imap_move_enable:  False
imap_move_folder:  getmail
imap_sync_folder:  INBOX
imap_ssl:          True
lmtp_hostname:     dovecot-mailcow
lmtp_port:         24
lmtp_debug:        True
lmtp_recipient:    mail@mail.local

#[example_insecure_test1_gmx.de]
#imap_hostname:     imap.gmx.net
#imap_port:         143
#imap_username:     example-test1@gmx.de
#imap_password:     xxxxx
#imap_ssl:          False

#[example_test2_gmx.de]
#imap_hostname:     imap.gmx.net
#imap_username:     example-test2@gmx.de
#imap_password:     xxxxx

[description-example-test-gmail-com]
imap_hostname:     imap.gmail.com
imap_username:     xxx@gmail.com
imap_password:     xxx
```
Account mail@mail.local exists in mailcow.

```
root@mail:/opt/getmail# docker network ls
NETWORK ID     NAME                                DRIVER    SCOPE
d885c6c5c3fa   bridge                              bridge    local
c0f159dd73c8   host                                host      local
d31bf24f0318   mailcowdockerized_mailcow-network   bridge    local
7659c559b0f6   network-getmail                     bridge    local
0aab0b6e7320   none                                null      local
```

```
root@mail:/opt/getmail# docker network inspect network-getmail
[
    {
        "Name": "network-getmail",
        "Id": "7659c559b0f69f62d183188f6b122e25849c24940c65b1c2443617cf16716758",
        "Created": "2022-06-06T07:13:10.133901678Z",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": true,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.23.11.0/24",
                    "Gateway": "172.23.11.1"
                },
                {
                    "Subnet": "fdcb:9c90:23:11::/64",
                    "Gateway": "fdcb:9c90:23:11::1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "058ba450d67a6da8585f128418a28a4e604ac98203c33fd952ea4c68a5c838cc": {
                "Name": "getmail_getmail_1",
                "EndpointID": "d2b79b788ec0aebba14069ee3a3df8b8c8256054ad6782f55524a9948547c7ee",
                "MacAddress": "02:42:ac:17:0b:03",
                "IPv4Address": "172.23.11.3/24",
                "IPv6Address": "fdcb:9c90:23:11::3/64"
            },
            "facc57fe186cb7ceccbc7751f6a52f67cd3f92cfe24a33ac8011f311b65e95f0": {
                "Name": "mailcowdockerized_dovecot-mailcow_1",
                "EndpointID": "65a223df776708059c5483d306154e74ac2993d9d1110846f687a726bc245476",
                "MacAddress": "02:42:ac:17:0b:02",
                "IPv4Address": "172.23.11.2/24",
                "IPv6Address": "fdcb:9c90:23:11::2/64"
            }
        },
        "Options": {
            "com.docker.network.bridge.name": "br-getmail"
        },
        "Labels": {}
    }
]
```

```
root@mail:/opt/getmail# cat /opt/mailcow-dockerized/docker-compose.override.yml 
version: '2.1'
services:

    dovecot-mailcow:
      networks:
        network-getmail:

networks:
  network-getmail:
    external: true

```
Any differences? Any misfire? Docker-compose steps not help. Dovecot logs from mailcow gui is empty, no any info about connection attempts.

**Comment by s4069b** [comment]
I had a look at each step and mine look the same - only difference I can see is in "docker network inspect network-getmail" 
my getmail_getmail_1 container comes after my mailcowdockerized_dovecot-mailcow_1 container, presumably because I started getmail after mailcow - don't know if that is important. 

I am not sure why you checked Dovecot logs in mailcow this is the log that you want to check
root@mail:/opt/getmail# docker-compose logs

If you do this:
  root@mail:/opt/getmail# docker-compose stop
then this:
  root@mail:/opt/getmail# docker-compose up -d
and then wait for say 20seconds and do this
  root@mail:/opt/getmail# docker-compose logs

You should see getmail connecting and pushing mail through to mailcow, if there is new mai in your google INBOX
LIke this
`getmail_1  | 2022-06-06 22:24:36,507 - Thread-name_domain.net - INFO: Login - status: b'name@domain.net authenticated (Success)'

getmail_1  | 2022-06-06 22:24:36,848 - Thread-name_domain.net - INFO: IMAP fetch mail - inital

getmail_1  | 2022-06-06 22:24:37,629 - Thread-name_domain.net - INFO: Join infinite loop and wait for new mails, cancel with Ctrl-c

getmail_1  | 2022-06-06 22:24:39,840 - Thread-name_domain.net-sent - ERROR: ERROR: [Errno -3] Try again

getmail_1  | 2022-06-06 22:24:39,841 - Thread-name_domain.net-sent - ERROR: ERROR: restart thread in 1 minutes (counter: 1)

getmail_1  | 2022-06-06 22:25:13,993 - Thread-name_domain.net - INFO: LMTP deliver: start -- LMTP host: dovecot-mailcow:24`

and on it goes

**Comment by byniow** [comment]
Because getmail if make connection to dovecot then dovecot should log this. I think.
I leaved germail working and now have 
```
INFO: Start Getmail - server: imap.gmail.com:993, username:xxx@gmail.com, ssl: True
getmail_1  | 2022-06-06 14:05:45,401 - Thread-description-example-test-gmail-com - INFO: Login - status: xxx@gmail.com authenticated (Success)
getmail_1  | 2022-06-06 14:05:55,572 - Thread-description-example-test-gmail-com - INFO: IMAP fetch mail - inital
getmail_1  | 2022-06-06 14:06:45,069 - Thread-description-example-test-gmail-com - ERROR: ERROR: command: FETCH => [OVERQUOTA] Account exceeded command or bandwidth limits.
getmail_1  | 2022-06-06 14:06:45,069 - Thread-description-example-test-gmail-com - ERROR: ERROR: restart thread in 1 minutes (counter: 1)
```
But fetchmail in iredmail correctly get mails via imap, thunderbird too without any problems.

**Comment by s4069b** [comment]
Yes, that's nothing like what I see - but it looks like a getmail issue - my guess would be a large attachment to an email or a large quantity of email wanting to be synced?  I have setup on an gmail account that is near empty, and have not tried receiving large files.

**Comment by byniow** [comment]
Today i add more accounts to sync to see what happen. 2 commercial and one private on iredmail (but other than mine). Now i have a lot new errors...
```
Attaching to getmail_getmail_1
getmail_1  | 2022-06-07 07:08:14,848 - MainThread - INFO: use config file: ./settings.ini
getmail_1  | 2022-06-07 07:08:19,850 - Thread-description-example-test-gmail-com - INFO: Start Getmail - server: imap.gmail.com:993, username:xxx@gmail.com, ssl: True
getmail_1  | 2022-06-07 07:08:19,850 - Thread-interia.pl - INFO: Start Getmail - server: poczta.interia.pl:993, username: xxx@interia.pl, ssl: True
getmail_1  | 2022-06-07 07:08:19,850 - Thread-tk - INFO: Start Getmail - server: post.xxx.xx:993, username: xxx@xxx.xx, ssl: True
getmail_1  | 2022-06-07 07:08:20,132 - Thread-interia.pl - INFO: Login - status: Success.
getmail_1  | 2022-06-07 07:08:20,153 - Thread-tk - INFO: Login - status: Logged in
getmail_1  | 2022-06-07 07:08:20,182 - Thread-tk - INFO: IMAP fetch mail - inital
getmail_1  | 2022-06-07 07:08:20,355 - Thread-interia.pl - INFO: IMAP fetch mail - inital
getmail_1  | 2022-06-07 07:08:21,003 - Thread-description-example-test-gmail-com - INFO: Login - status: xxx@gmail.com authenticated (Success)
getmail_1  | 2022-06-07 07:08:21,093 - Thread-interia.pl - ERROR: ERROR: command: FETCH => Internal error.
getmail_1  | 2022-06-07 07:08:21,094 - Thread-interia.pl - ERROR: ERROR: restart thread in 1 minutes (counter: 1)
getmail_1  | 2022-06-07 07:08:31,185 - Thread-description-example-test-gmail-com - INFO: IMAP fetch mail - inital
getmail_1  | 2022-06-07 07:08:51,511 - Thread-tk - INFO: LMTP deliver: start -- LMTP host: dovecot-mailcow:24
getmail_1  | 2022-06-07 07:08:51,527 - Thread-tk - ERROR: LMTP deliver (ConnectionRefusedError): [Errno 111] Connection refused
getmail_1  | 2022-06-07 07:08:51,527 - Thread-tk - INFO: LMTP deliver: start -- LMTP host: dovecot-mailcow:24
getmail_1  | 2022-06-07 07:08:51,528 - Thread-tk - ERROR: LMTP deliver (ConnectionRefusedError): [Errno 111] Connection refused
```

Gmail have 500 mails (~3GB), other less. So, when getmail finally login and want fetch mails (thread-tk - my iredmail server) then lmtp in mailcow have problem. Any other config in mailcow needed?

**Comment by Sieboldianus** [comment]
.. just jumping in here @byniow @s4069b

Could you please encompass your big code and output text in `<details><summary></summary></details>` tags?

Otherwise the issue here (and other issues) become impossible to follow.

<details><summary>Example</summary>

```
example code
```

</details>

**Comment by christianbur** [comment]
The error message "[OVERQUOTA] Account exceeded command or bandwidth limits." comes from the imap.gmail.com server and has nothing to do with getmail. Please move the current emails in the INBOX to another folder (for testing), maybe there are too many emails in the INBOX. The goal of Getmail is to empty the whole INBOX.

**Comment by christianbur** [comment]
The error message "[OVERQUOTA] Account exceeded command or bandwidth limits." comes from the imap.gmail.com server and has nothing to do with getmail. Please move the current emails in the INBOX to another folder (for testing), maybe there are too many emails in the INBOX. The goal of Getmail is to empty the whole INBOX. 

Something seems to be wrong with the connection to Mailcow/Dovecot "ERROR: LMTP deliver (ConnectionRefusedError): [Errno 111] Connection refused"

**Comment by byniow** [comment]
Yes, i have the same behavior when i set mailcow fetcher every 1 minute. Fetchmail every 1 minute via imap not have any problems, when i use imapsync then all mails is correctly fetched without any limits. And - if getmail not connecting with lmtp, not synchronizing mails why limits are exceed?
About LMTP - this is fresh install, nothing changed in mailcow except add domain and mailbox. Propably is any little step (or steps) clear to You but not clear for rest....

If i can make any tests or logs tell me, i try paste any needed info.

EDIT:
Ok, i cleared gmail INBOX and getmail show `Join infinite loop and wait for new mails, cancel with Ctrl-c`
What with thread interia.pl? This is Polish free mail provider. 
Ill try discover why LMTP not work.

**Comment by christianbur** [comment]
Now you have to put an email into the INBOX of gmail and see what happens in the log of getmail. 
I have been using getmail with gmail for many years without any problems. 

you can also test if "dovecot-mailcow" is pingable from getmail, that should work. 
```
#docker exec -it getmail_getmail_1 ping dovecot-mailcow
or
#docker exec -it getmail-getmail-1 ping dovecot-mailcow

PING dovecot-mailcow (172.23.11.3): 56 data bytes
64 bytes from 172.23.11.3: seq=0 ttl=42 time=0.054 ms
64 bytes from 172.23.11.3: seq=1 ttl=42 time=0.060 ms
64 bytes from 172.23.11.3: seq=2 ttl=42 time=0.069 ms
```

**Comment by byniow** [comment]
Hi, i tested mails in gmail and getmail see it, but lmtp error showed like on tk thread.

```
root@mail:/opt/getmail# docker exec -it getmail_getmail_1 ping dovecot-mailcow
PING dovecot-mailcow (172.23.11.2): 56 data bytes
ping: permission denied (are you root?)
```

**Comment by byniow** [comment]
I add` iputils` to Dockerfile in RUN section. And now ping work:
```
root@mail:/opt/getmail# docker exec -it getmail_getmail_1 ping dovecot-mailcow
PING dovecot-mailcow (172.23.11.3) 56(84) bytes of data.
64 bytes from mailcowdockerized_dovecot-mailcow_1.network-getmail (172.23.11.3): icmp_seq=1 ttl=64 time=0.107 ms
64 bytes from mailcowdockerized_dovecot-mailcow_1.network-getmail (172.23.11.3): icmp_seq=2 ttl=64 time=0.122 ms
```
But LMTP delivier errors still exist.

**Comment by byniow** [comment]
https://github.com/mailcow/mailcow-dockerized/issues/3279
Main problem with lmtp is in mailcow. In link above i find info about ipv6 problems with connectivity inside mailcow. Errors in this thread i have identical like in my installation. I disable ipv6, restart mailcow and getmail start working. 

And, interia.pl start working after clear INBOX. But what strange - iRedMail account with ~200 mails has been correctly fetched by getmail without clearing INBOX. 

Thanks for help.
